pages/api/expenses/[id]/index.js

```diff
@@ -481,4 +481,7 @@ export default async function handler(req, res) {
 
     return res.status(200).send();
   }
+
+  res.setHeader('Allow', ['POST', 'GET', 'PUT', 'DELETE']);
+  return res.status(405).end(`Method ${method} Not Allowed`);
```

Comment:
Trying out Next.js' example from their v11 docs. I feel like explicitly setting the allowed methods HAS to work.

https://github.com/vercel/next.js/blob/v11.1.2/examples/api-routes-rest/pages/api/user/%5Bid%5D.js